modules/runtime/nuxeo-runtime-launcher/src/main/java/org/nuxeo/log4j2/MaskSensitiveDataRewritePolicy.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * (C) Copyright 2021 Nuxeo (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Kevin Leturc <kleturc@nuxeo.com>
+ */
+
+package org.nuxeo.log4j2;
+
+import java.util.regex.Pattern;
+
+import org.apache.logging.log4j.core.Core;
+import org.apache.logging.log4j.core.LogEvent;
+import org.apache.logging.log4j.core.appender.rewrite.RewritePolicy;
+import org.apache.logging.log4j.core.config.plugins.Plugin;
+import org.apache.logging.log4j.core.config.plugins.PluginFactory;
+import org.apache.logging.log4j.core.impl.Log4jLogEvent;
+import org.apache.logging.log4j.message.Message;
+import org.apache.logging.log4j.message.SimpleMessage;
+
+/**
+ * @since 11.5
+ */
+@Plugin(name = "MaskSensitiveData", category = Core.CATEGORY_NAME, elementType = "rewritePolicy")
+public final class MaskSensitiveDataRewritePolicy implements RewritePolicy {
+
+    protected static final Pattern AWS_KEY_PATTERN = Pattern.compile(
+            "(A3T[A-Z0-9]|AKIA|AGPA|AIDA|AROA|AIPA|ANPA|ANVA|ASIA)([A-Z0-9]{3})[A-Z0-9]{9}([A-Z0-9]{4})");
+
+    // cards can only start with 4, 5, 6, 34, 35 or 37
+    protected static final Pattern CREDIT_CARD_PATTERN = Pattern.compile(
+            "(([456][0-9]{3}|3[457][0-9]{2})-?[0-9]{4}-?[0-9]{4}-?[0-9]{4})");
+
+    protected static final Pattern GCP_KEY_PATTERN = Pattern.compile(
```

Comment:
Please document where the pattern comes from as well

Reply:
This was given by IT. I found this repository which contains some regex https://github.com/odomojuli/RegExAPI, do we want to reference it?

Reply:
Maybe in the ticket then. But it's worth asking IT where they got this from.